actions/test/run_local.ts

```diff
@@ -55,14 +60,92 @@ const main = async () => {
       }
     });
+  } else if (contractAddressEnv) {
+    // If no blockNumberEnv is provided, then we rebuild the state from the default deployment block
+    if (!blockNumberEnv) {
+      console.log(`[run_rebuild] No block number provided, using default deployment block`);
+    } else {
+      assert(!isNaN(Number(blockNumberEnv)), "blockNumber must be a number");
+    }
+    let fromBlock = blockNumberEnv ? Number(blockNumberEnv) : DEFAULT_DEPLOYMENT_BLOCK;
+
+    // Rebuild the state
+    console.log(`[run_rebuild] Rebuild the state of the conditional orders from the historical events.`);
+    const contractAddress = process.env.CONTRACT_ADDRESS;
+    assert(contractAddress && ethers.utils.isAddress(contractAddress), "contract address is required");
+    const pageSize = process.env.PAGE_SIZE ? parseInt(process.env.PAGE_SIZE) : DEFAULT_PAGE_SIZE;
+
+    // 1. Record the current block number - useful with paging
+    let currentBlockNumber = await provider.getBlockNumber();
+    console.log(`[run_rebuild] Current block number: ${currentBlockNumber}`);
+
+    // 2. Connect to the contract instance
+    const contract = ComposableCoW__factory.connect(contractAddress, provider);
+    
+    // 3. Define the filter.
+    const filter = contract.filters.ConditionalOrderCreated();
+
+    // 4. Get the historical events
+    const replayPlan: ReplayPlan = {}
+    let toBlock;
+    do {
+      toBlock = (pageSize == 0) ? 'latest' : fromBlock + (pageSize - 1);
+      if (typeof toBlock !== 'string' && toBlock > currentBlockNumber) {
+          // refresh the current block number
+          currentBlockNumber = await provider.getBlockNumber();
+          toBlock = (toBlock > currentBlockNumber) ? currentBlockNumber : toBlock;
+
+          console.log(`[run_rebuild] Reaching tip of chain, current block number: ${currentBlockNumber}`);
+      }
+
+      console.log(`[run_rebuild] Processing events from block ${fromBlock} to block ${toBlock}`);
+
+      const events = await contract.queryFilter(filter, fromBlock, toBlock);
+
+      console.log(`[run_rebuild] Found ${events.length} events`);
+
+      // 5. Process the events
+      for (const event of events) {
+        if (replayPlan[event.blockNumber] === undefined) {
+          replayPlan[event.blockNumber] = new Set();
+        }
+
+        replayPlan[event.blockNumber].add(event.transactionHash);
+      }
+
+      // only possible string value for toBlock is 'latest'
+      if (typeof toBlock !== 'string') {
+          fromBlock = toBlock + 1;
+      }
+    } while (toBlock !== 'latest' && (typeof toBlock !== 'string' && toBlock != currentBlockNumber));
+
+    // 6. Replay the blocks by iterating over the replayPlan
+    for (const [blockNumber, txHints] of Object.entries(replayPlan)) {
+      console.log(`[run_rebuild] Processing block ${blockNumber}`);
+      const overrides: ProcessBlockOverrides = {
+        blockWatchBlockNumber: currentBlockNumber,
+        txList: Array.from(txHints)
```

Comment:
I think that this is not needed if you update to my changes in #32 

In this PR I allow to process one single TX, and that's what you are doing here. 
It feels strange to artificially "process the block" filtering by the TX you want, instead of processing directly the TXs:

https://github.com/cowprotocol/tenderly-watch-tower/pull/32/files#diff-a9c764fec942e612065756fc4d17975445bb87affca07113b249d45d8d9fb090R106